interceptors_idempotency.go

```diff
@@ -175,18 +185,20 @@ func (i *idempotencyInterceptor) controllerPublishVolume(
 	}
 	defer lock.Unlock()
 
-	// At the end of this function check for a response error. If there
-	// is an error that isn't VolumeNotFound then mark this method as
-	// in error so the next call bypasses idempotency.
-	defer func() {
-		handleErr(info, lock, resErr)
-	}()
-
-	// If the method has been marked in error then it means a previous
-	// call to this function returned an error. In these cases a
-	// subsequent call should bypass idempotency.
-	if _, ok := lock.methodInErr[info.FullMethod]; ok {
-		return handler(ctx, req)
+	if i.opts.cacheErrs {
+		// At the end of this function check for a response error. If there
+		// is an error that isn't VolumeNotFound then mark this method as
```

Comment:
I should know this, but why the stated exception for `VolumeNotFound`? If a controllerPublishVolume call returns VolumeNotFound, it seems like if the call is retried, we would want to bypass idempotency and defer to the SP.

Reply:
Hi @codenrhoden,

Primarily to be consistent with the rest of it, but also because the debt of **not** bypassing idempotency for this is rather small since it will work by determining the volume does not exist.